src/hardware/mixer.cpp

```diff
@@ -705,8 +734,12 @@ void MixerChannel::SetReverbLevel(const float level)
 	assert(level >= level_min);
 	assert(level <= level_max);
 
-	if (!HasFeature(ChannelFeature::ReverbSend))
+	do_reverb_send = (HasFeature(ChannelFeature::ReverbSend) && level > FLT_EPSILON);
```

Comment:
`FLT_EPSILON` doesn't work like that -- it's smallest difference between two successive representable float numbers around zero (so there's only a single-bit difference in the binary representations of those two float values). You'd never want to use it for epsilon-comparisons, it serves a different purpose.

Yes, it's a very bad idea to compare floats with plain equality when you're dealing with floats that are results of some calculations or physical measurements; in those cases you always want to do the epsilon-style comparison, indeed. However, the exact epsilon to use is always very specific to the nature of those calculations or measurements, there's simply no general epsilon that would work in all cases. You either need to calculate the exact error range mathematically based on what calculations you're performing, or use some fudge factor (most people do the latter). Plus you'd need to use a relative difference formula in the epislon comparison, which is a lot more involved.

The exception to all this is when you _know_ for certain that the float contains some integers. That's what we're dealing here -- it's ensured that the value never goes below zero (because it's clamped), and when the reverb is off, it's set to 0.0f. Therer's absolutely no uncertainty there, the binary representation of 0.0f is fixed and never changes.

Integers are represented perfectly up to very large numbers when stored as floating-point numbers (you can look up the exact limits for various precisions; it's a bit below the int32 & int64 limits, but still very large), so it you put in 0.0, 1.0, 2.0, 3.0, etc., you will always get out 0.0, 1.0, 2.0, 3.0 on every hardware that supports IEEE floats, no exceptions, it's ensured by the standard.

Short story: please just use the `> 0.0f` comparison everywhere, that's the simplest thing to do, and it's sufficient for our purposes here.

Reply:
Thanks for the very thorough explanation, @johnnovak ; especially:

> exception to all this is when you know for certain that the float contains some integers. That's what we're dealing here -- it's ensured that the value never goes below zero (because it's clamped), and when the reverb is off, it's set to 0.0f. Therer's absolutely no uncertainty there.

Will drop this 👍